Core2/Onboarding/src/main/java/com/infomaniak/library/onboarding/OnboardingScaffold.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Infomaniak SwissTransfer - Android
+ * Copyright (C) 2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.library.onboarding
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.pager.HorizontalPager
+import androidx.compose.foundation.pager.PagerState
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.ImageVector.Builder
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalLayoutDirection
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun OnboardingScaffold(
+    pagerState: PagerState,
+    onboardingPages: List<OnboardingPage>,
+    bottomContent: @Composable (PaddingValues) -> Unit,
+) {
+    Scaffold { paddingValues ->
+        Column {
+            val left = paddingValues.calculateLeftPadding(LocalLayoutDirection.current)
+            val right = paddingValues.calculateRightPadding(LocalLayoutDirection.current)
+
+            HorizontalPager(pagerState, modifier = Modifier.weight(1f)) {
+                OnboardingPageContent(
+                    modifier = Modifier.windowInsetsPadding(
+                        WindowInsets(left = left, top = paddingValues.calculateTopPadding(), right = right)
+                    ),
+                    page = onboardingPages[it],
+                    0.dp,
+                )
+            }
+
+            HorizontalPagerIndicator(
```

Comment:
I figured, when you're in landscape mode you're already struggling a lot with vertical space which is why I was thinking, I'd rather not put vertical margins to not loose more space. Instead of having something nice at least you can see a bit more info on the screen

Reply:
Finally we decided to lock the screen in portrait mode for small device to avoid these problems